src/Common/MetricHelper.cs

```diff
@@ -508,7 +508,7 @@ private void PopulateConfigMetricsFromQos(AzurePSQoSEvent qos, IDictionary<strin
             {
                 foreach (var configMetric in qos.ConfigMetrics)
                 {
-                    eventProperties[configMetric.ConfigKey] = configMetric.ConfigValue;
+                    eventProperties[configMetric.Value.ConfigKey] = configMetric.Value.ConfigValue;
```

Comment:
It can be confusing that ConfigKey doesn't necessary mean the ConfigKey. Consider adding a property to ConfigMetrics to represent the telemetry key.